server/api/controller/plan.js

```diff
@@ -73,7 +76,50 @@ class PlanController extends Controller {
 			q.where.geo_search_filter = [false];
 		}
 
-		return super.browse(req, q);
+		return super.browse(req, q, this.afterFetch);
+	}
+
+
+	// attached tag_name into tags. The initial fetch brings only tags id, but we want the tags name.
+	async afterFetch (collection) {
+
+		// take all the unique tag ids, and get it into an array (set doesn't work in the db query later on)
+		const relevantTagIds = Array.from(new Set(collection.models.map(planModel => {
```

Comment:
Hi @TomerMe2 can you please take a look on that link? I think bookshelf allows this functionality internally. 
https://bookshelfjs.org/api.html#Model-instance-related

Reply:
Seems that it can work, but it will require a db query for each plan_tag instance.
The docs says that it works on a specific model and not on a collection. Do you want me to refactor into using model.related()? I can do that, but I think that it's not the correct solution for our problem.

Reply:
That's an approach that'd enbale us to fetch all relevant models at 1 query rather than in twice